src/PrestaShopBundle/Resources/config/services/core/form/choice_provider.yml

```diff
@@ -227,3 +227,13 @@ services:
 
   prestashop.core.form.choice_provider.address_required_fields:
     class: 'PrestaShop\PrestaShop\Core\Form\ChoiceProvider\AddressRequiredFieldsChoiceProvider'
+
+  prestashop.core.form.choice_provider.customer_service_order_messages_name:
+    class: 'PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CustomerServiceOrderMessagesNameChoiceProvider'
+    arguments:
+      - '@=service("prestashop.adapter.order_message.order_message_provider").getMessages()'
```

Comment:
I'm a bit worried about this.
1) When is this call performed ? At Symfony booting time (dev and prod environments ?) ? If yes, it means we're doing a potentially heavy SQL search at booting time, for all requests, not only the ones for this page. Can you check ?
2) In prod env, the container is dumped. How is this being dumped in php ? I'm worried that SF would first run the `getMessages()` function then inject the serialized result into CustomerServiceOrderMessagesNameChoiceProvider making it a static result instead of a dynamic one. Can you check ?

Reply:
OK, I checked, it's alright 👍